client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -150,20 +150,19 @@ private PositionInternal getPosition() {
     @GuardedBy("readers")
     private String updateGroupStateIfNeeded() throws ReinitializationRequiredException {
         try {
+            if (atCheckpoint != null) {
+                groupState.checkpoint(atCheckpoint, getPosition());
```

Comment:
Observe that `fetchUpdatesIfNeeded()` is not invoked before the state is updated with `CheckpointReader(...)`. As per my current understanding it should not cause an issue. (just wanted to highlight minor difference)

Reply:
Good point. Yes. It will not cause any issue.